cpp/inc/bond/ext/grpc/detail/service.h

```diff
@@ -95,26 +105,56 @@ class service : private grpc::Service
         return this;
     }
 
+    Scheduler& scheduler()
+    {
+        return _scheduler;
+    }
+
 protected:
-    service() = default;
+    template <typename MethodT>
+    using Method = service_unary_call_data<
+        typename MethodT::input_type,
+        typename remove_bonded<
+            typename std::conditional<
+                std::is_void<typename MethodT::result_type>::value,
+                Void,
+                typename MethodT::result_type>::type>::type>;
+
+    service(const Scheduler& scheduler, std::initializer_list<const char*> methodNames)
+        : _scheduler(scheduler ? scheduler : thread_pool{}),
+          _cq(nullptr)
+    {
+        AddMethods(methodNames);
+    }
 
-    /// @brief Registers a method name for dispatch to this service.
-    ///
-    /// @note This method is for use by generated and helper code only.
-    ///
-    /// The order in which methods are registered assigned the method index,
-    /// which is used elsewhere.
-    void AddMethod(const char* methodName)
+private:
+    friend class gRPC::server_builder;
+
+    void AddMethods(std::initializer_list<const char*> names)
     {
-        BOOST_ASSERT(methodName);
-
-        // ownership of the service method is transfered to grpc::Service
-        grpc::Service::AddMethod(
-            new grpc::internal::RpcServiceMethod(
-                methodName,
-                grpc::internal::RpcMethod::NORMAL_RPC,
-                nullptr)); // nullptr indicates async handler
+        for (const char* name : names)
+        {
+            BOOST_ASSERT(name);
+
+            // ownership of the service method is transfered to grpc::Service
+            grpc::Service::AddMethod(
+                new grpc::internal::RpcServiceMethod(
+                    name,
+                    grpc::internal::RpcMethod::NORMAL_RPC,
+                    nullptr)); // nullptr indicates async handler
+        }
     }
+
+    void SetCompletionQueue(grpc::ServerCompletionQueue* cq)
+    {
+        BOOST_ASSERT(!_cq);
+        _cq = cq;
+    }
+
+    Scheduler _scheduler;
+    grpc::ServerCompletionQueue* _cq;
 };
 
-} } } } //namespace bond::ext::gRPC::detail
+} } } } // namespace bond::ext::gRPC::detail
+
+#include "service_call_data.h"
```

Comment:
This `#include` change goes away with https://github.com/Microsoft/bond/pull/893, right?

Reply:
Yes.